packages/flutter/lib/src/material/slider.dart

```diff
@@ -703,13 +707,14 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
         color: theme.colorScheme.onPrimary,
       ),
     );
+    final Set<MaterialState> states = <MaterialState>{
+      if (!_enabled) MaterialState.disabled,
+      if (_hovering) MaterialState.hovered,
+      if (_focused) MaterialState.focused,
+    };
     final MouseCursor effectiveMouseCursor = MaterialStateProperty.resolveAs<MouseCursor>(
-      widget.mouseCursor ?? MaterialStateMouseCursor.clickable,
-      <MaterialState>{
-        if (!_enabled) MaterialState.disabled,
-        if (_hovering) MaterialState.hovered,
-        if (_focused) MaterialState.focused,
-      },
+      widget.mouseCursor ?? sliderTheme.mouseCursor?.resolve(states) ?? MaterialStateMouseCursor.clickable,
```

Comment:
This is pretty good but not quite right. If `widget.mouseCursor` resolves to null for the current state, then we want to use the theme's mouse cursor, which may also resolve to null for the current state. So:
```dart
MaterialStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)  ??
MaterialStateProperty.resolveAs<MouseCursor?>(sliderTheme.mouseCursor, states) ?? 
MaterialStateMouseCursor.clickable
```

Reply:
Oh, right. Thanks!